File: valid-anagram/Anagram.java
import java.util.HashMap;
import java.util.Scanner;

public class Anagram {
    /**
     * Given two strings s and t, return true if s is an anagram of t, and otherwise return false
     * 
     * @param s first array of strings to be compared
     * @param t second array of strings to be compared
     * @return True if the two strings are anagram, else false
     */

    public static boolean checkAnagram(String s, String t) {
        if (s.length() != t.length())
            return false;

        HashMap<Character, Integer> count = new HashMap<>();

        for (char c : s.toCharArray()) {
            count.put(c, count.getOrDefault(c, 0) + 1);
        }

        for (char c : t.toCharArray()) {
            if (!count.containsKey(c)) {
                return false;
            }
            count.put(c, count.getOrDefault(c, 0) + 1);
            if (count.get(c) < 0) {
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter the first word: ");
        String input1 = scanner.nextLine();

        System.out.print("Enter the Second word: ");
        String input2 = scanner.nextLine();
        scanner.close();

        String anagram1 = input1.trim().toLowerCase();
        String anagram2 = input2.trim().toLowerCase();

        if (checkAnagram(anagram1, anagram2)) {
            System.out.println("true");
        } else {
            System.out.println("false");
        }
    }
}

File: valid-anagram/Anagram.cpp
#include <string>
#include <iostream>
#include <unordered_map>

/**
 * Given two strings s and t, return true if s is an anagram of t, and otherwise return false
 *
 * @param s first array of strings to be compared
 * @param t second array of strings to be compared
 * @return True if the two strings are anagram, else false
 */

std::string cleanString(std::string str)
{
    str.erase(std::remove_if(str.begin(), str.end(), ::isspace), str.end());
    std::transform(str.begin(), str.end(), str.begin(), ::tolower);

    return str;
}

bool checkAnagram(const std::string &s, const std::string &t)
{
    if (s.length() != t.length())
        return false;

    std::unordered_map<char, int> count;
    for (char c : s)
        count[(unsigned char)c]++;

    for (char c : t)
    {
        if (--count[c] < 0)
            return false;
    }

    return true;
}

int main()
{
    std::cout << "Enter the first word: ";
    std::string input1;
    std::getline(std::cin, input1);

    std::cout << "Enter the second word: ";
    std::string input2;
    std::getline(std::cin, input2);

    input1 = cleanString(input1);
    input2 = cleanString(input2);

    if (checkAnagram(input1, input2))
    {
        std::cout << "true" << std::endl;
    }
    else
    {
        std::cout << "false" << std::endl;
    }

    return 0;
}